File: jarvis/src/integrationTest/java/com/mikhalochkin/jarvis/TestService.java
package com.mikhalochkin.jarvis;

import org.apache.commons.io.IOUtils;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.io.InputStream;
import java.nio.charset.StandardCharsets;
import java.util.Optional;

/**
 * Service for integration tests.
 *
 * @author Alex Mikhalochkin
 */
@Service
public class TestService {

    /**
     * Returns content of the file.
     *
     * @param pathToFile path to the file with the payload.
     * @return content of the file.
     */
    public String getPayloadFromFile(String pathToFile) {
        return Optional.ofNullable(pathToFile)
                .map(this.getClass().getClassLoader()::getResourceAsStream)
                .map(this::getContent)
                .orElseThrow(() -> new RuntimeException(String.format("File [%s] not found", pathToFile)));
    }

    private String getContent(InputStream stream) {
        try {
            return IOUtils.toString(stream, StandardCharsets.UTF_8.name());
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: momomo-app/src/testIntegration/kotlin/com/am/jarvis/TestService.kt
package com.am.jarvis

import java.io.FileNotFoundException
import java.nio.charset.StandardCharsets
import java.util.Locale
import java.util.Optional
import org.apache.commons.io.IOUtils
import org.springframework.stereotype.Service

/**
 * Service for integration tests.
 *
 * @author Alex Mikhalochkin
 */
@Service
class TestService {

    /**
     * Returns content of the file.
     *
     * @param pathToFile path to the file with the payload.
     * @return content of the file.
     */
    fun getPayloadFromFile(pathToFile: String): String {
        return Optional.ofNullable(pathToFile)
            .map { this.javaClass.classLoader.getResourceAsStream(it) }
            .map { IOUtils.toString(it, StandardCharsets.UTF_8.name()) }
            .orElseThrow {
                FileNotFoundException(String.format(Locale.getDefault(), "File [%s] not found", pathToFile))
            }
    }
}
